Origin/feature/inventory UI

radnonymous opened the pull request:
Creates inventory UI menu and weight tracker

radnonymous committed:
initial commit

radnonymous committed:
Merge branch 'develop' into origin/feature/inventoryUI

radnonymous committed:
initial commit

radnonymous committed:
Weight UI

getmikyled commented on CannibalisticZombies/Assets/Assets.index:
```diff
@@ -0,0 +1,13 @@
+{
+    "type": "scene",
```
Do you know what these changes are? They are not one that I recognize

radnonymous replied:
No, I'm not sure either. I'll remove the file edit from the PR.

getmikyled commented on CannibalisticZombies/Assets/Scripts/UI/InventoryBackground.cs:
```diff
@@ -0,0 +1,22 @@
+    // Start is called before the first frame update
+    void Start()
+    {
+        this.transform.parent = transform;
```
What is this script for? Also I am a little confused lol is this setting its parent to itself?

getmikyled replied:
If this line of code is needed, lets do it inside the regular `InventoryUI `script. You can just do `[SerializeField]` to access the background and run the code in there.

radnonymous replied:
Is not needed, removed

getmikyled commented on CannibalisticZombies/Assets/Scripts/UI/InventoryUI.cs:
```diff
@@ -0,0 +1,155 @@
+
+            // construct and space line
+            finalLine = firstHalf;
+            for (int i = 0; i < LINELENGTH - firstHalf.Length - secondHalf.Length; i += spacing.Length) 
```
How accurate is this? It seems that in the screenshot the alignment might be off. Could be because the spacing string " -" contains two characters which could mess up the math?

radnonymous replied:
The for loop iterated by the length of the " -" spacing string, so that shouldn't be a problem. The reason the inventory is "centered" and not "justified" is because there is a fixed line character length, but not all characters are the same width.

getmikyled commented on CannibalisticZombies/Assets/Scripts/UI/InventoryUI.cs:
```diff
@@ -0,0 +1,155 @@
+{
+    public class InventoryUI : MonoBehaviour
+    {
+        public Inventory inventory;
```
I do not think we would want to access all of these methods publicly, as all inventory shenanigans relating to UI should be done in this script. For all the variables that this applies to below (Not including `LINELENGTH`), lets use `[SerializeField] private .........`. We do not want to mark things public unless they are to be accessed by other scripts.

getmikyled reviewed (commented):
Done leaving feedback! Overall looks good. Can you also make it so that the inventory appears and disappears when you press tab? Thanks for your hardwork Ross!

radnonymous committed:
Adjustments

Removed InventoryBackground script
Made WriteSlotEntry for loop more intuitive (start at firstHalf length instead of 0)
updated InventoryUI class variables to private
updated secondHalf for nicer spacing

radnonymous commented:
I'm having an issue where the Serialized Fields for the Background image game objects do not show up on the editor, so i cannot connect them to the InventoryUI script (needed to toggle their visibility with Tab) do you also have this issue? Is there a way to address this?

radnonymous committed:
Implement Menu Toggle Method
